src/main/scala/com/microsoft/hyperspace/index/rules/JoinIndexRule.scala

```diff
@@ -305,8 +305,8 @@ object JoinIndexRule
     val rRequiredIndexedCols = lRMap.values.toSeq
 
     // All required columns resolved with base relation.
-    val lRequiredAllCols = resolve(spark, allRequiredCols(left), leftRelation.plan).get.map(_._1)
-    val rRequiredAllCols = resolve(spark, allRequiredCols(right), rightRelation.plan).get.map(_._1)
+    val lRequiredAllCols = resolve(spark, allRequiredCols(left), lBaseAttrs).get
+    val rRequiredAllCols = resolve(spark, allRequiredCols(right), rBaseAttrs).get
```

Comment:
I am reverting this change. This can be reintroduced when `JoinIndexRule` is updated.